common/ClientRequestCleaner.js

```diff
@@ -0,0 +1,19 @@
+const RequestsCleaner = require('./RequestsCleaner');
+
+class ClientRequestsCleaner extends RequestsCleaner {
+	verifySignatures() {
+		if (!this.calls) {
+			return;
+		}
+		Object.keys(this.calls).forEach((procedure) => {
+			Object.keys(this.calls[procedure]).forEach((signature) => {
+				if (RequestsCleaner.isOutdated(signature, this.timeoutMs)) {
+					this.calls[procedure][signature].reject('RPC response timeout exceeded');
```

Comment:
What's your current error and exception handling strategy?

In this specific case a string is OK because you don't really need a stack trace for that, BUT if it is possible for other parts of the code to call the `reject` and pass in `Error` instance (instead of a string) then the handler will now have to deal with two different potential error types (both strings and Error objects) which is inconsistent and adds complexity to the error handlers themselves.

Also, sometimes you want a simple way to identify errors (like an `error.name` or `error.code`) when doing comparisons. Relying on the message can be problematic in the long run: `if (error == 'RPC response timeout exceeded')` is not as nice/futureproof as `if (error.code == 'RPC_RESPONSE_TIMEOUT')`. Node.js itself is moving towards adding a `code` property to internal error objects to decouple the error message from its error code.

It's important that the type of error object/string passed to the `reject` is consistent. If you decide to go with strings all the time, then you need to be sure that they are sufficient for your needs. There are ways to dehydrate errors to pass them across process boundaries and reconstruct the stack trace (if that is useful at all?). Some projects like to just throw/emit/reject using Objects instead of Errors or strings because they are easier to pass across client/server and process boundaries.

Just something to think about.